### PR TITLE
Update to match sample app user

### PR DESCRIPTION
**Pull request by wiktork**

### CODE REVIEW

File: documentation/kubernetes.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ In addition to its availability as a .NET CLI tool, the `dotnet monitor` tool is
 
 For Dockerfiles and repository information, see [Running in Docker](./docker.md)
 
+## Non-root considerations
+
+Starting with .NET 8.0, both the sample aspnet application and dotnet-monitor run as non-root. If both the application and dotnet-monitor are 8+, no additional configuration is required. Otherwise, a [security context](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) configuration may need to be added to the application, dotnet-monitor, or both.
```
Comment (wiktork) [inline documentation/kubernetes.md:12]: @poppastring I can remove or shorten this section and move to any additional non-root docs we have

File: documentation/kubernetes.md [hunk 2]
```diff
@@ -49,6 +53,11 @@ spec:
             memory: 512Mi
       - name: monitor
         image: mcr.microsoft.com/dotnet/monitor:6
+        securityContext:
+          # Default APP_UID for non-root dotnet application images
+          runAsUser: 64198
```
Comment (wiktork) [inline documentation/kubernetes.md:58]: Use 1654, latest UID

File: documentation/kubernetes.md [hunk 3]
```diff
@@ -152,6 +166,75 @@ spec:
...
+  name: deploy-exampleapp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: exampleapp
+  template:
+    metadata:
+      labels:
+        app: exampleapp
```
Comment (wiktork) [inline documentation/kubernetes.md:186]: Also update our other deployment sample

File: documentation/kubernetes.md [hunk 4]
```diff
@@ -152,6 +166,75 @@ spec:
...
+    spec:
+      restartPolicy: Always
+      containers:
```
Comment (wiktork) [inline documentation/kubernetes.md:189]: Per @schmittjoseph's testing, add fsGroup for volume mounted configuration examples.

File: documentation/kubernetes.md [hunk 5]
```diff
@@ -152,6 +166,75 @@ spec:
...
+      - name: app
+        image: mcr.microsoft.com/dotnet/samples:aspnetapp
+        imagePullPolicy: Always
+        env:
+        - name: ASPNETCORE_URLS
+          value: http://+:80
```
Comment (wiktork) [inline documentation/kubernetes.md:195]: Per @schmittjoseph's testing, cannot bind to port 80 as non-root

Comment (jander-msft) [reply documentation/kubernetes.md:195]: 8080 is used in the non-root .NET Docker images by default

**Pull request merged**